client.go

```diff
@@ -50,28 +50,41 @@ func (c *client) DialBack(ctx context.Context, p peer.ID) (ma.Multiaddr, error)
 	err = w.WriteMsg(req)
 	if err != nil {
 		s.Reset()
-		return nil, err
+		return nil, nil, err
 	}
 
 	var res pb.Message
 	err = r.ReadMsg(&res)
 	if err != nil {
 		s.Reset()
-		return nil, err
+		return nil, nil, err
 	}
 
 	if res.GetType() != pb.Message_DIAL_RESPONSE {
-		return nil, fmt.Errorf("Unexpected response: %s", res.GetType().String())
+		return nil, nil, fmt.Errorf("Unexpected response: %s", res.GetType().String())
 	}
 
 	status := res.GetDialResponse().GetStatus()
```

Comment:
can we call `res.GetDialResponse()` once, and use status / successAddrs / FailedAddrs off that object below?